From 794d80629983244606b25f3c799d6424d3a7e4bd Mon Sep 17 00:00:00 2001
From: Hemant Kumar <hemantk@codeaurora.org>
Date: Tue, 3 Jun 2014 00:08:17 -0700
Subject: [PATCH] xhci: hsic: Fix NULL pointer dereference in
 mxhci_hsic_hub_control

Primary HCD pointer is set when initializing SS HCD structure.
There is possibility of hub_port_staus getting called for HS
root hub when primary HCD pointer is not initialized. Hence
do not access primary HCD pointer in mxhci_hsic_hub_control.

Change-Id: Ie91a87b1a2d6757410f1f5c9c6010e2099c29420
CRs-Fixed: 674442
Signed-off-by: Hemant Kumar <hemantk@codeaurora.org>
---
 drivers/usb/host/xhci-msm-hsic.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/host/xhci-msm-hsic.c b/drivers/usb/host/xhci-msm-hsic.c
index 838b79e..dc1a711 100644
--- a/drivers/usb/host/xhci-msm-hsic.c
+++ b/drivers/usb/host/xhci-msm-hsic.c
@@ -970,12 +970,16 @@ void mxhci_hsic_shutdown(struct usb_hcd *hcd)
 int mxhci_hsic_hub_control(struct usb_hcd *hcd, u16 typeReq, u16 wValue,
 		u16 wIndex, char *buf, u16 wLength)
 {
-	struct mxhci_hsic_hcd *mxhci = hcd_to_hsic(hcd->primary_hcd);
+	struct mxhci_hsic_hcd *mxhci;
 	int ret = 0;
 	u32 status;
 
 	ret = xhci_hub_control(hcd, typeReq, wValue, wIndex, buf, wLength);
 
+	if (!hcd->primary_hcd)
+		return ret;
+
+	mxhci = hcd_to_hsic(hcd->primary_hcd);
 	status = get_unaligned_le32(buf);
 
 	if (typeReq == GetPortStatus) {
-- 
1.7.8.3

